src/software/pipeline/main_relativePoseEstimating.cpp

```diff
@@ -0,0 +1,402 @@
+
+#include <aliceVision/sfmData/SfMData.hpp>
+#include <aliceVision/sfmDataIO/sfmDataIO.hpp>
+#include <aliceVision/sfm/pipeline/regionsIO.hpp>
+#include <aliceVision/feature/imageDescriberCommon.hpp>
+
+#include <aliceVision/track/TracksBuilder.hpp>
+#include <aliceVision/track/tracksUtils.hpp>
+#include <aliceVision/track/trackIO.hpp>
+
+#include <aliceVision/camera/Pinhole.hpp>
+
+#include <aliceVision/robustEstimation/ACRansac.hpp>
+#include <aliceVision/multiview/essential.hpp>
+#include <aliceVision/multiview/relativePose/Essential5PSolver.hpp>
+#include <aliceVision/multiview/relativePose/FundamentalError.hpp>
+#include <aliceVision/multiview/RelativePoseKernel.hpp>
+#include <aliceVision/multiview/triangulation/triangulationDLT.hpp>
+
+#include <aliceVision/sfm/liealgebra.hpp>
+
+#include <boost/program_options.hpp>
+#include <boost/filesystem.hpp>
+
+#include <aliceVision/sfm/pipeline/relativePoses.hpp>
+
+#include <cstdlib>
+
+// These constants define the current software version.
+// They must be updated when the command line is changed.
+#define ALICEVISION_SOFTWARE_VERSION_MAJOR 1
+#define ALICEVISION_SOFTWARE_VERSION_MINOR 0
+
+using namespace aliceVision;
+
+namespace po = boost::program_options;
+namespace fs = boost::filesystem;
+
+
+bool getPoseStructure(Mat3& R, Vec3& t, std::vector<Vec3>& structure, std::vector<size_t>& newVecInliers, const Mat3& E,
+                      const std::vector<size_t>& vecInliers, const Mat3& K1, const Mat3& K2, const Mat& x1,
+                      const Mat& x2)
+{
+    // Find set of analytical solutions
+    std::vector<Mat3> Rs;
+    std::vector<Vec3> ts;
+    motionFromEssential(E, &Rs, &ts);
+
+    Mat34 P1, P2;
+    Mat3 R1 = Mat3::Identity();
+    Vec3 t1 = Vec3::Zero();
+    P_from_KRt(K1, R1, t1, &P1);
+
+    size_t bestCoundValid = 0;
+
+    for(int it = 0; it < Rs.size(); it++)
+    {
+        const Mat3& R2 = Rs[it];
+        const Vec3& t2 = ts[it];
+
+        P_from_KRt(K2, R2, t2, &P2);
+
+        std::vector<Vec3> points;
+        std::vector<size_t> updatedInliers;
+
+        size_t countValid = 0;
+        for(size_t k = 0; k < vecInliers.size(); ++k)
+        {
+            const Vec2& pt1 = x1.col(vecInliers[k]);
+            const Vec2& pt2 = x2.col(vecInliers[k]);
+
+            Vec3 X;
+            multiview::TriangulateDLT(P1, pt1, P2, pt2, &X);
+
+            // Test if point is front to the two cameras.
+            if(Depth(R1, t1, X) > 0 && Depth(R2, t2, X) > 0)
+            {
+                countValid++;
+            }
+
+            updatedInliers.push_back(vecInliers[k]);
+            points.push_back(X);
+        }
+
+        if(countValid > bestCoundValid)
+        {
+            bestCoundValid = countValid;
+            structure = points;
+            newVecInliers = updatedInliers;
+            R = Rs[it];
+            t = ts[it];
+        }
+    }
+
+    if(newVecInliers.size() < 10)
+    {
+        return false;
+    }
+
+    return true;
```

Comment:
```suggestion
    return newVecInliers.size() >= 10;
```
Maybe `10` can be a parameter?